tests/test.py

```diff
@@ -102,6 +103,26 @@ def tearDownClass(cls):
         """
         invoke.run(cls.dbt_clean)
 
+    def compare_files(self, path1, path2):
+        # using filecmp will not work properly across different os
+        with open(path1) as f1:
+            for idx1, line1 in enumerate(f1):
+                with open(path2) as f2:
+                    for idx2, line2 in enumerate(f2):
+                        if idx2 != idx1:
+                            continue
+                        else:
+                            # matching line1 from both files
+                            if line1 == line2:
+                                break
+                            else:
+                                self.logger.info(
+                                    f"Found mismatching line: \n{line1}\n{line2}"
+                                )
+                                # else print that line from both files
+                                return False
+            return True
```

Comment:
It seems like it takes a bit of work to find the first mismatching line.  Do you feel that we need it?

I was thinking of alternatives:
- We're already dealing with pathlib Path objects so we could just check if `path1.read_text() == path2.read_text()`
  - It's true that this doesn't tell us which is the first offending line. If we really want that we can do something like:
  ```
  path1_text = path1.read_text()
  path2_text = path2.read_text()
  path1_lines = path1_text.splitlines()
  path2_lines = path2_text.splitlines()
  zipped_lines = itertools.zip_longest(path1_lines, path2_lines)
  for i, (path1_line, path2_line) in enumerate(zipped_lines):
      if path1_line != path2_line:
          self.logger.info(f"Found mismatch in line {i+1}: \n{path1_line}\n{path2_line}")
          return False
  return True
  ```
  - There's also [difflib](https://docs.python.org/3/library/difflib.html) in the standard library if we feel we want to display the whole diff as opposed to the first diff.

Reply:
I think I was trying to avoid having both entire files in memory but model files are highly unlikely of being of super large size (and if so, I guess you have a different problem).
So your solution is much prettier, I agree.

Reply:
Also, I think the first mismatching line is fine here. In the end, the output will only be used to debug the test...